### ISSUE TITLE
[GradScaler] Inconsistent scale values across different GPUs caused by uneven inputs for AMP DDP training

### ISSUE THREAD

**Issue by JiJiJiang** [pytorch/pytorch#90057]
### 🐛 Describe the bug

Hello, I want to add [Automatic Mixed Precision (AMP)](https://pytorch.org/docs/master/amp.html?highlight=gradscaler#) training into my DDP training codes.

In my case, different GPUs could have uneven inputs during the training process. Thus I use model.join() to handle this problem and it works.

However, to realize the AMP training, I use the [torch.cuda.amp.GradScaler](https://pytorch.org/docs/master/amp.html#torch.cuda.amp.GradScaler) class. Due to the uneven inputs problem, we get inconsistent scale values after several epochs, which gradually causes a large loss difference and training error. 

For example, 

1. Due to data imbalance, rank 0 could run more batches/iterations using model.join().
![image](https://user-images.githubusercontent.com/23723621/205296877-7ae6b562-b65d-4dee-8e81-bb11aba2ed2f.png)

2. Since growth_interval is same (default: 2000) for rank 0 and 1, after several training epochs, rank 0 and 1 could run different numbers of batchs/iterations, which leads to different growth timestamp of scale. As shown below, at Epoch 17 Iter 2071, they got different scale values.
![image](https://user-images.githubusercontent.com/23723621/205297947-800daa15-5af4-42d6-9d5e-ff956f35c760.png)
![image](https://user-images.githubusercontent.com/23723621/205296993-10019b59-9c7a-4f1e-8870-bdc98ea6cec5.png)

I think this could be a bug of torch.cuda.amp.GradScaler while using uneven inputs for DDP training. 
**There should be some synchronization mechanism to keep the consistency of the scale value (or the growth_tracker) across different GPUs in the case of uneven inputs.**

Thanks a lot if you can help answer my question!

### Versions

pytorch 1.12.1
2 * A100

**Comment by JiJiJiang** [comment]
Any progress?
